rust/private/rustc.bzl

```diff
@@ -714,12 +714,14 @@ def construct_arguments(
     if hasattr(attr, "_error_format"):
         rustc_flags.add("--error-format=" + attr._error_format[ErrorFormatInfo].error_format)
 
-    # Mangle symbols to disambiguate crates with the same name
-    extra_filename = "-" + output_hash if output_hash else ""
-    rustc_flags.add("--codegen=metadata=" + extra_filename)
+    if output_hash:
+        # Mangle symbols to disambiguate crates with the same name
+        extra_filename = "-" + output_hash
+        rustc_flags.add("--codegen=metadata=" + extra_filename)
```

Comment:
I'm a bit confused by this gating. Can you add a comment about why these two flags are now conditionally added?

Reply:
added some comments